modules/apps/frontend-js/frontend-js-dropdown-support-web/src/main/resources/META-INF/resources/DropdownProvider.js

```diff
@@ -106,12 +97,20 @@ class DropdownProvider {
 		trigger.parentElement.classList.add(CssClass.SHOW);
 		trigger.setAttribute('aria-expanded', true);
 
-		document.addEventListener('mousedown', (event) =>
-			this._maybeClickOutside(event, {menu, trigger})
-		);
-		document.addEventListener('touchstart', (event) =>
-			this._maybeClickOutside(event, {menu, trigger})
-		);
+		const clickOutsideHandler = (event) => {
+			if (
+				!menu.contains(event.target) &&
+				!trigger.container(event.target)
+			) {
+				this.hide({menu, trigger});
+
+				document.removeEventListener('mousedown', clickOutsideHandler);
+				document.removeEventListener('touchstart', clickOutsideHandler);
+			}
+		};
+
+		document.addEventListener('mousedown', clickOutsideHandler);
+		document.addEventListener('touchstart', clickOutsideHandler);
```

Comment:
If it was just `mousedown`, you could simplify this with
```js
Liferay.once('mousedown', clickOutsideHandler);
```
That way you wouldn't have to `removeEventListener`.

I am not sure how two listeners with the same handler work with `once`.

Reply:
I could test it. Thanks Marko :)

Reply:
Nevermind, ignore me, this needs to trigger on click everywhere :)

Reply:
![image](https://media.giphy.com/media/fDO2Nk0ImzvvW/giphy.gif)

Reply:
Hey @markocikos, I don't mind using Liferay's global event handler for in-app events(events which aren't to have an inter-widget-communication, for example notifying a widget from another) but I can test it

Reply:
When using like:
```js
		const clickOutsideHandler = (event) => {
			if (
				!menu.contains(event.target) &&
				!trigger.contains(event.target)
			) {
				this.hide({menu, trigger});
			}
		};

		Liferay.once('mousedown', clickOutsideHandler);
		Liferay.once('touchstart', clickOutsideHandler);
```

the handler isn't being called :/

Reply:
when removing `Liferay.once('touchstart', clickOutsideHandler);` it still don't working :/

Reply:
`Liferay.once` and `Liferay.on` AFAIK are to listen to events emited by `Liferay.fire` so no, that won't work

Reply:
Github is trolling me, notifications were just updated when hard reloading the page 🤦‍♂️